magma/backend/mlir/when.py

```diff
@@ -0,0 +1,241 @@
+import contextlib
+import itertools
+
+from magma.backend.mlir.builtin import builtin
+from magma.backend.mlir.hw import hw
+from magma.backend.mlir.magma_common import (
+    visit_value_or_value_wrapper_by_direction as
+    visit_magma_value_or_value_wrapper_by_direction,
+)
+from magma.backend.mlir.mlir import get_block_stack, MlirValue, push_block
+from magma.backend.mlir.sv import sv
+from magma.common import sort_by_value
+from magma.primitives.when import iswhen
+from magma.ref import ArrayRef, TupleRef
+from magma.value_utils import make_selector
+
+
+def _assert_false(*args, **kwargs):
```

Comment:
can this be removed from the other file? if not, let's move this to `magma/common.py`